### PR TITLE
releasenote cleanup

### PR DESCRIPTION
**Pull request by spiffxp**
We got rid of the old release-note-label-needed

The release note docs moved on us, and based on a convo with 1.11 release team folks about release notes, I don't think many people look in the "Details" section.

### CODE REVIEW

**Comment by spiffxp** [comment]
/hold
for comment

File: prow/plugins/releasenote/releasenote.go [hunk 1]
```diff
@@ -43,16 +43,13 @@ const (
 	releaseNoteNone           = "release-note-none"
 	releaseNoteActionRequired = "release-note-action-required"
 
-	releaseNoteFormat       = `Adding %s because the release note process has not been followed.`
-	releaseNoteSuffixFormat = `One of the following labels is required %q, %q, or %q.
-Please see: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed.`
+	releaseNoteFormat       = `Adding the "%s" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/contributors/guide/release-notes.md) to remove it.`
```
Comment (cblecker) [inline prow/plugins/releasenote/releasenote.go]: broken link. one too many contributors/ in it

Comment (spiffxp) [reply prow/plugins/releasenote/releasenote.go]: 🤦‍♂️  good catch

**Pull request merged**